runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5195,6 +5195,21 @@ TR_J9ByteCodeIlGenerator::loadInstance(TR::SymbolReference * symRef)
          }
       }
 
+   static char *disableFinalFieldFoldingInILGen = feGetEnv("TR_DisableFinalFieldFoldingInILGen");
+   static char *disableInstanceFinalFieldFoldingInILGen = feGetEnv("TR_DisableInstanceFinalFieldFoldingInILGen");
+   if (!disableFinalFieldFoldingInILGen &&
+       !disableInstanceFinalFieldFoldingInILGen &&
+       address->getOpCode().hasSymbolReference() &&
+       address->getSymbolReference()->hasKnownObjectIndex() &&
+       address->isNonNull())
+      {
+      TR::Node* nodeToRemove = NULL;
+      if (TR::TransformUtil::transformIndirectLoadChain(comp(), dummyLoad, address, address->getSymbolReference()->getKnownObjectIndex(), &nodeToRemove) && nodeToRemove)
```

Comment:
will this make sure we protect the point with an appropriate set of OSR guards etc in fear analysis?

Reply:
The fields folded here don't need protection. They're fields trusted by VM and changes on them are not expected.